WordPress/gutenberg
# [RNMobile] Update Slider to reflect changes immediately

## Description
This PR fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2192
gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2286

In this PR i use `onValueChange ` instead of `onSlidingComplete ` to set the value.

<details>
 <summary>Outdated</summary>
Unfortunately in the current state, it is laggy because of performance issues in block-list. (The whole app is re-rendered on every change). I tried these changes with that PR (https://github.com/WordPress/gutenberg/pull/22144) - and it works much better. I suggest merging the PR with optimization first.
</details>


I need to NOTE that it is not a performant way of doing that kind of thing in React Native since the communication is async and we can lose frames. Especially in our case, because there are a lot of things that happen on each change of attribute. We shouldn't rely on the value from redux when we "animate" the height etc. It doesn't mean that we shouldn't use `onValueChange` instead of `onSlidingComplete` but we should change how it is handled in Block. As an example, In the spacer block, we could rely on the local state and set the attribute with a debounce effect. Thanks to that the height should be more responsive because right now I can feel the delay.



There is also an issue with the undo mechanism that I described here - https://github.com/wordpress-mobile/gutenberg-mobile/issues/2192#issuecomment-630010164

This issue was fixed, however it introduced a regression and had to be reverted.

## How has this been tested?
- open mobile app
1.
- add spacer block
- open settings of the spacer block
- change the height of spacer - change should be visible while moving the slider
- change the height a few time more
- press on the undo button - the height should be reset to initial one
- the undo level should be created only when we change a different attribute
2.
- add button block
- open settings of that button
- change the border-radius -  change should be visible while moving the slider
- the undo level should be created only when we change a different attribute
3.
- add spacer block and open settings
- change the input value to less than minimum value i.e `2`
- the slider should show the minimum value (20)
- the height of the block should be changed to the minimum value (20)
- there should be a possibility to change the input value
- after return, if the input value is lower than the minimum value then is set to minimum and the same with the maximum limit

<details>
 <summary>Outdated</summary>
In the current state, the undo level is created for every change that means 10 undo levels will be created when sliding from 0 to 10 (https://github.com/wordpress-mobile/gutenberg-mobile/issues/2192#issuecomment-630010164)
</details>



## Screenshots 

without optimization | with optimimsation
--- | ---
![slideiosbefore](https://user-images.githubusercontent.com/16336501/82226295-9e306480-9926-11ea-8456-aeaa4bb5c76f.gif) | ![sliderios](https://user-images.githubusercontent.com/16336501/82226375-b99b6f80-9926-11ea-86f9-67da356e6f0a.gif)

## Types of changes
Update Range Control to reflect changes immediately

## Checklist:
